File: shipping-service/src/domain/event/order_event.rs
use serde::{Deserialize, Serialize};

use crate::domain::enums::e_order_event::EOrderEvent;

#[derive(Debug, Serialize, Deserialize)]
pub struct OrderEvent {
    #[serde(rename(serialize = "eventType", deserialize = "eventType"))]
    pub event_type: String,
    #[serde(rename(serialize = "orderId", deserialize = "orderId"))]
    pub order_id: i64,
}

impl OrderEvent {
    pub fn is_valid_to_consume(&self) -> bool {
        if self.event_type == EOrderEvent::to_string(&EOrderEvent::OrderAccepted) {
            return true;
        };
        return false;
    }
}

File: payment-service/src/domain/event/order_event.py
from domain.enums.e_order_event_type import EOrderEventType


class OrderEvent:
    def __init__(self, eventType: str, orderId: int):
        self.event_type = eventType
        self.order_id = orderId

    def isValidToConsume(self):
        return self.event_type == EOrderEventType.INVENTORY_RESERVED.name
